### PR TITLE
Documentation related to settings_build and settings_target

### PR DESCRIPTION
**Pull request by jgsogo**
Docs for https://github.com/conan-io/conan/pull/6769 and https://github.com/conan-io/conan/pull/6916

### CODE REVIEW

File: devtools/build_requires.rst [hunk 1]
```diff
@@ -69,7 +69,7 @@ Build requirements can also be specified in a package recipe, with the ``build_r
 The above ``tool_a`` and ``tool_b`` will always be retrieved and used for building this recipe, while the ``tool_win`` one will only be used
 only in Windows.
 
-If some build requirement defined inside ``build_requirements()`` has the same package name as the one defined in the ``build_requires``
+If some build requirement defined inside ``build_requirements()`` have the same package name as the one defined in the ``build_requires``
```
Comment (memsharded) [inline devtools/build_requires.rst:72]: This was correct, ``build requirement`` is singular

File: systems_cross_building/cross_building.rst [hunk 2]
```diff
@@ -193,6 +193,46 @@ The above means that **Conan is able to compile the full graph in a single execu
...
+        def configure(self):
+            settings_target = getattr(self, 'settings_target', None)
+            if settings_target is None:
+                # It is running in 'host', so Conan is compiling this package
+                if not self.options.target:
+                    raise ConanInvalidConfiguration("A value for option 'target' has to be provided")
+            else:
+                # It is running in 'build' and it is being used as a BR, 'target' can be inferred from settings
+                if self.options.target:
+                    raise ConanInvalidConfiguration(f"Value for the option 'target' will be computed from settings_target")
```
Comment (memsharded) [inline systems_cross_building/cross_building.rst:232]: f-string?

File: systems_cross_building/cross_building.rst [hunk 3]
```diff
@@ -193,6 +193,46 @@ The above means that **Conan is able to compile the full graph in a single execu
...
+  packages in the ``host`` context, this attribute will contain the settins from the profile ``profile_host``, while
+  for recipes that are build requirements of other build requirements the ``self.settings_target`` will contain
+  the values of the ``profile_build``.
+
+With previous attributes, a draft for a recipe that packages a cross compiler could follow this pattern:
+
+.. code-block:: python
+
+    class CrossCompiler(ConanFile):
+        name = "xbuild"
```
Comment (memsharded) [inline systems_cross_building/cross_building.rst:217]: Call it "mycompiler"?

File: systems_cross_building/cross_building.rst [hunk 4]
```diff
@@ -193,6 +193,46 @@ The above means that **Conan is able to compile the full graph in a single execu
...
+            settings_target = getattr(self, 'settings_target', None)
+            if settings_target is None:
+                # It is running in 'host', so Conan is compiling this package
+                if not self.options.target:
+                    raise ConanInvalidConfiguration("A value for option 'target' has to be provided")
+            else:
+                # It is running in 'build' and it is being used as a BR, 'target' can be inferred from settings
+                if self.options.target:
+                    raise ConanInvalidConfiguration(f"Value for the option 'target' will be computed from settings_target")
+                self.options.target = get_target_from_settings(self.settings_target)
```
Comment (memsharded) [inline systems_cross_building/cross_building.rst:233]: This ``get_target_from_settings()`` is a bit confusing. If it is an option, I would expect to be in the input. Where is it defined? What it does?

Comment (jgsogo) [reply systems_cross_building/cross_building.rst:233]: It is a function to _get the target [value] from the settings_. I'll write this function before the class with a comment. 

This is the cross compiler example from the repo in https://github.com/jgsogo/conan-xbuild, when this recipe is instantiated in the `build` context then the value for the `target` option CAN be computed from the profile_host (which is the value of `settings_target`).

**Pull request merged**